src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,190 @@
+package duke;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.InvalidPathException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * The storage class is in charge of recording the user's input and loading saved user record.
+ * Saving and loading tasks are done automatically by default, but a user can save and load the tasks to a specified
+ * directory by using the save/load command.
+ */
+public class Storage {
+    private enum Month {
+        JANUARY, FEBRUARY, MARCH, APRIL, MAY, JUNE, JULY,
+        AUGUST, SEPTEMBER, OCTOBER, NOVEMBER, DECEMBER
+    }
+    private ArrayList<Task> userInputRecords;
+
+    /**
+     * The constructor for a Storage Object.
+     */
+    public Storage() {
+        this.userInputRecords = new ArrayList<>();
+    }
+
+    /**
+     * Automatically(without any user command) loads the users inputs from ../data/record when the program is restarted.
+     */
+    public void autoLoad() {
+        try {
+            Scanner scanner = new Scanner(Paths.get("data", "record"));
+            while (scanner.hasNextLine()) {
+                String itemInfo = scanner.nextLine();
+                if (itemInfo.startsWith("[T]")) {
+                    Task task = new ToDo(itemInfo.substring(7));
+                    if (itemInfo.contains("[X]")) {
+                        task.setDone(true);
+                    }
+                    userInputRecords.add(task);
+                } else if (itemInfo.startsWith("[D]")) {
+                    Task task = new Deadline(itemInfo.substring(7, itemInfo.indexOf("(by")),
+                            convertToLocalTime(itemInfo.substring(itemInfo.indexOf("(by") + 5,
+                                    itemInfo.length() - 1)));
+                    if (itemInfo.contains("[X]")) {
+                        task.setDone(true);
+                    }
+                    userInputRecords.add(task);
+                } else if (itemInfo.startsWith("[E]")) {
+                    Task task = new Event(itemInfo.substring(7, itemInfo.indexOf("(at")),
+                            convertToLocalTime(itemInfo.substring(itemInfo.indexOf("(at") + 5,
+                                    itemInfo.length() - 1)));
+                    if (itemInfo.contains("[X]")) {
+                        task.setDone(true);
+                    }
+                    userInputRecords.add(task);
+                }
+            }
+            scanner.close();
+        } catch (IOException e) {
+            System.out.println("Saved data not found, a new data file created.\n");
+        }
+    }
```

Comment:
Perhaps you can abstract or simplify this method more. It's quite hard to follow the logic behind this with so much with the nested while and if(s) loops.

Reply:
Thanks for the suggestion! I have abstracted out the add task process to another private method in the latest commit!